pipelined/bevy_render2/src/lib.rs

```diff
@@ -84,17 +84,21 @@ struct ScratchRenderWorld(World);
 
 impl Plugin for RenderPlugin {
     fn build(&self, app: &mut App) {
-        let (instance, device, queue) =
-            futures_lite::future::block_on(renderer::initialize_renderer(
-                BackendBit::PRIMARY,
-                &wgpu::RequestAdapterOptions {
-                    power_preference: wgpu::PowerPreference::HighPerformance,
-                    ..Default::default()
-                },
-                &wgpu::DeviceDescriptor::default(),
-            ));
-        app.insert_resource(device.clone())
-            .insert_resource(queue.clone())
+        let renderer = app
+            .world
+            .remove_resource::<renderer::Renderer>()
```

Comment:
This is to remove the `Renderer` resource if it exists right?

I would prefer an explicit check / branch over a removal here.

Reply:
It'll create ownership problems if we leave `renderer::Renderer` in the world. Specifically, `RenderInstance` is not `Clone`. So, you can have `RenderInstance` in the app or the render subapp, but not both.